libglusterfs/src/mem-pool.c

```diff
@@ -91,35 +91,28 @@ gf_mem_set_acct_info(struct mem_acct *mem_acct, struct mem_header *header,
                      size_t size, uint32_t type, const char *typestr)
 {
     struct mem_acct_rec *rec = NULL;
-    bool new_ref = false;
+    uint64_t num_allocs;
 
     if (mem_acct != NULL) {
         GF_ASSERT(type <= mem_acct->num_types);
 
         rec = &mem_acct->rec[type];
+        if (!rec->typestr) {
+            rec->typestr = typestr;
+        }
+        num_allocs = GF_ATOMIC_INC(rec->num_allocs);
+        rec->max_size = max(rec->max_size, (num_allocs * size));
+        rec->max_num_allocs = max(rec->max_num_allocs, num_allocs);
+        if (num_allocs == 1)
+            GF_ATOMIC_INC(mem_acct->refcnt);
```

Comment:
Is it safe to assume that when a xlator terminates all its memory should already be released ? if so we don't need this reference.

The only problematic thing I'm aware of is the garbage collector used for mem pools, but now that we don't use them, are there other places where "frees" can be delayed after the termination of the owner xlator ?

Reply:
At cli level memory cleanup are not correctly handled, as we know all cli processes are instantly started/stopped so we don't care to cleanup the the memory so we have to keep it.